### ISSUE TITLE
Unfork lsp-types

### ISSUE THREAD

**Issue by ismail** [astral-sh/ruff#11595]
Looks like https://github.com/gluon-lang/lsp-types has released version 0.96.0 with notebook support, would it be possible to use it instead of using a custom fork?

**Comment by MichaReiser** [comment]
Yeah, I think that should be possible. All we needed was that the notebook PR gets merged and released and that's now the case.

**Comment by MichaReiser** [comment]
Hmm, that sounds painful. I don't think we're in a rush updating. Let's see what they decide on.

**Comment by T-256** [comment]
cc @karthiknadig who announced `lsprotocol` crate [at here](https://github.com/astral-sh/ruff-lsp/issues/300#issuecomment-1813500582) and was looking for feedback. I recommend you try `ruff_server` migrated to `lsprotocol`. I tried but there were few blockers that stopped me on continue it. you can catch these blockers as feedback to improve `lsprotocol` crate.

**Comment by snowsignal** [comment]
The issue blocking an upgrade to 0.96.0, as @T-256 pointed out, is https://github.com/gluon-lang/lsp-types/issues/284. We use the old `Url` type frequently in the codebase and the new `Uri` type lacks several key functions that we need, especially related to file path conversion.

I would be interested in exploring a move to `lsprotocol`, though as @T-256 said there could be blockers there too.

**Comment by dhruvmanila** [comment]
> I tried but there were few blockers that stopped me on continue it.

Can you enumerate the blockers here? We'd appreciate that. It could also be useful to provide the diff here and anyone from the team can enumerate the said blockers.

**Comment by dhruvmanila** [comment]
This isn't required now that we've switched to a different crate in https://github.com/astral-sh/ruff/pull/24894.
